phoenix-core/src/test/java/org/apache/phoenix/compile/WhereCompilerTest.java

```diff
@@ -183,7 +183,7 @@ public void testSingleFixedFullPkSalted() throws SQLException {
     @Test
     public void testSingleVariableFullPkSalted() throws SQLException {
         PhoenixConnection pconn = DriverManager.getConnection(getUrl(), PropertiesUtil.deepCopy(TEST_PROPERTIES)).unwrap(PhoenixConnection.class);
-        pconn.createStatement().execute("CREATE TABLE t (k varchar primary key, v varchar) SALT_BUCKETS=20");
+        pconn.createStatement().execute("CREATE TABLE t (k varchar(10) primary key, v varchar) SALT_BUCKETS=20");
```

Comment:
Looks like this issue would exist for any variable-length field and not just varbinary. Can you update the JIRA subject and description accordingly?

Reply:
So for any other field varchar doesn't really have valid characters near the 0 byte so the scan ranges of ['joe','joe\0) and ['joe','joe\1) are equivalent.   The other variable length field at risk is maybe arraytype?  Do we allow arraytype in a PK?

Reply:
As sycned offline, I'll add an arraytype E2E test as this does affect any variable-length field and I believe that I can construct a failing E2E for it unlike varchar.  I will also update the PR/Jira/Commit text to reflect.